Arrowgene.Ddon.GameServer/Handler/InstanceEnemyKillHandler.cs

```diff
@@ -31,32 +31,38 @@ public InstanceEnemyKillHandler(DdonGameServer server) : base(server)
             _gameServer = server;
         }
 
-        public override void Handle(GameClient client, StructurePacket<C2SInstanceEnemyKillReq> packet)
+        public override S2CInstanceEnemyKillRes Handle(GameClient client, C2SInstanceEnemyKillReq packet)
         {
-            CDataStageLayoutId layoutId = packet.Structure.LayoutId;
+            CDataStageLayoutId layoutId = packet.LayoutId;
             StageId stageId = StageId.FromStageLayoutId(layoutId);
 
             // The training room uses special handling to produce enemies that don't exist in the QuestState or InstanceEnemyManager.
             // Return an empty response here to not break the rest of the handling.
             if (_ignoreKillsInStageIds.Contains(stageId.Id))
             {
-                client.Send(new S2CInstanceEnemyKillRes());
-                return;
+                return new();
             }
 
             Quest quest = null;
             bool IsQuestControlled = false;
             foreach (var questScheduleId in client.Party.QuestState.StageQuests(stageId))
             {
                 quest = client.Party.QuestState.GetQuest(questScheduleId);
-                if (client.Party.QuestState.HasEnemiesInCurrentStageGroup(quest, stageId))
+                if (quest != null && client.Party.QuestState.HasEnemiesInCurrentStageGroup(quest, stageId))
```

Comment:
This is a very strange check. How do we have quests in the current quest state but not have a quest for it?